libfreerdp/crypto/x509_utils.c

```diff
@@ -663,46 +659,42 @@ static BYTE* x509_utils_get_pem(const X509* xcert, const STACK_OF(X509) * chain,
 		}
 	}
 
-	offset = 0;
-	length = 2048;
+	const size_t blocksize = 2048;
+	size_t offset = 0;
+	size_t length = blocksize;
 	pemCert = (BYTE*)malloc(length + 1);
 
-	if (!pemCert)
+	if (!pemCert || (length > INT_MAX))
 	{
 		WLog_ERR(TAG, "error allocating pemCert");
 		goto fail;
 	}
 
-	ERR_clear_error();
-	status = BIO_read(bio, pemCert, length);
-
-	if (status < 0)
+	while (offset < length)
```

Comment:
Wouldn't it be better to ditch the loop altogether and just use `BIO_pending` to query the necessary buffer size?

Reply:
actually, on a better look, the whole function is unused :grin: